### PR TITLE
[CSSWA-451] Openshift 4.6 removed features

### PR DESCRIPTION
**Pull request by gmcrocetti**
* Removed `OperatorSource` from OperatorInstaller task - API removed >= 4.6
* Remove `CatalogSource` from OperatorInstaller task - API removed >= 4.5
* Remove _source_processor method of OperatorInstaller's task
* Refactor to use f-strings
* Also solves [CSSWA-356]

### CODE REVIEW

File: piqe_ocp_lib/api/resources/ocp_operators.py [hunk 1]
```diff
@@ -205,6 +209,12 @@ class OperatorSource(OcpBase):
     :param kube_config_file: A kubernetes config file.
     :return: None
     """
+
```
Comment (gmcrocetti) [inline piqe_ocp_lib/api/resources/ocp_operators.py:212]: CSSWA-356

File: piqe_ocp_lib/api/resources/ocp_projects.py [hunk 2]
```diff
@@ -134,9 +134,8 @@ def does_project_exist(self, project_name):
         :param project_name:
         :return: True if the project is found. False if the project is not found.
         """
```
Comment (gmcrocetti) [inline piqe_ocp_lib/api/resources/ocp_projects.py:136]: pythonic

File: piqe_ocp_lib/tests/resources/test_ocp_operators.py [hunk 3]
```diff
@@ -256,66 +283,42 @@ class TestInstallOperatorWorkflow:
...
-        get_test_objects.project_obj.delete_a_project('test-project7')
+            assert cluster_service.is_cluster_service_version_present(csv_name, project.metadata.name)
+
+        project_resource.delete_a_project('test-amq-streams-allnamespaces-og-sub-project')
+        project_resource.delete_a_project('test-project7')
         # TODO: look into why the csv persits in openshift-operators project after cleaning
         # test artifacts
-        get_test_objects.project_obj.delete_a_project('openshift-operators')
-
-    @pytest.mark.skip(reason="Skip until MPQEENABLE-356 is resolved")
```
Comment (gmcrocetti) [inline piqe_ocp_lib/tests/resources/test_ocp_operators.py:297]: Keep tasks compliant with openshift's latest version, for now.

**Review by amacmurr — APPROVED** [review]
Looks good to me. Ship it!

File: piqe_ocp_lib/api/tasks/operator_ops.py [hunk 4]
```diff
@@ -1,9 +1,8 @@
 from piqe_ocp_lib.api.resources import OcpBase, OcpProjects
 from piqe_ocp_lib.api.resources.ocp_operators import OperatorhubPackages, \
-    Subscription, CatalogSource, OperatorSource, OperatorGroup
```
[the remark this answers is not in the document]
Comment (gmcrocetti) [reply piqe_ocp_lib/api/tasks/operator_ops.py:3]: oh yeah, we can definitely solve this. Here's a quick solution:
```python
#  abstraction to all specific version installers
class BaseInstaller:
    
    def install_operator(self):
        raise NotImplementedError
    
    def remove_operator(self):
        raise NotImplementedError

#  concrete implementation for version 3.x
class OCP3Installer(BaseInstaller):

    def install_operator(self):
        # concrete implementation
        pass


#  Refactoring our current implementation
class InstallerTask(BaseTask):
    # ....

    @cached_property
    def _installer(self, cluster_info):
        if cluster_info.version <= '3.0.0':
            return OCP3Installer()

    def install_operator(self, *args, **kwargs):
        self._installer.install_operator(args, kwargs)
    
```

File: piqe_ocp_lib/tests/helpers.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+from piqe_ocp_lib.api.resources import OcpClusterVersion
+
+
+#  Change to dataclass (requires py >= 3.7)
+class Config:
+    def __init__(self, kubeconfig: Optional[str] = None):
+        self.kubeconfig = kubeconfig or os.environ.get("KUBECONFIG")
+
+    @property  # Change to cached_property (py >= 3.8)
+    def version(self) -> Tuple[int, int, int]:
```
Comment (btjd) [inline piqe_ocp_lib/tests/helpers.py:16]: Might want to rename this property to cluster_version. My first reaction was to think that this is ocp version. Note that ocp_base has a 'version' property for the product version. Now that I think of it, are 'version' on ocp_base and what get_cluster_version redundant?

File: piqe_ocp_lib/api/tasks/operator_ops.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (btjd) [reply piqe_ocp_lib/api/tasks/operator_ops.py:3]: Pretty much what I had in mind

File: piqe_ocp_lib/tests/helpers.py (hunk 5 above)
Comment (gmcrocetti) [reply piqe_ocp_lib/tests/helpers.py:16]: Might be worth refactoring in favor of `OcpBase`

**Comment by gmcrocetti** [comment]
@btjd , @amacmurr . Could you please take a second look ? I've refactored `version` in OcpBase

**Comment by amacmurr** [comment]
The CI failure is due an issue in the backing infrastructure and is not code related. This PR may be merged.

**Review by btjd — APPROVED** [review]

**Comment by btjd** [comment]
LGTM, shitp it!

**Pull request merged**